jsplot/src/main/java/tech/tablesaw/plotly/api/BubblePlot.java

```diff
@@ -38,22 +42,43 @@ public static Figure create(
   }
 
   public static Figure create(
-      String title, Table table, String xCol, String yCol, String sizeColumn) {
-    Layout layout = Layout.builder(title, xCol, yCol).build();
-
-    Marker marker =
-        Marker.builder()
-            .size(table.numberColumn(sizeColumn))
-            // .opacity(.75)
-            .build();
-
-    ScatterTrace trace =
-        ScatterTrace.builder(table.numberColumn(xCol), table.numberColumn(yCol))
-            .marker(marker)
-            .build();
+      String title,
+      Column xColumn,
+      Column yColumn,
+      NumericColumn sizeColumn,
+      double[] color,
+      SizeMode sizeMode,
+      Double opacity) {
+
+    Layout layout = Layout.builder(title, xColumn.name(), yColumn.name()).build();
+
+    Marker marker = null;
+    MarkerBuilder builder = Marker.builder();
+    if (sizeColumn != null) {
```

Comment:
But you should add a quick javadoc about which parameters are nullable and which not. For example here sizeColumn is nullable, but in the other method sizeCol cannot be null.